[Refactor] lombok 적용

MyLittleChicken opened the pull request:
## 개요📜
entity 클래스에 lombok 을 적용하여 코드의 가독성과 유지보수성을 개선하였습니다.
## 수정사항 🛠️
- Lombok 의존성 추가
- ShortenUrlEntity 개선
    - getter, constructor 에 대한 어노테이션 적용
## 기대효과 
- 코드 간결화: 반복적인 코드를 제거할 수 있게 됩니다.
- 가독성 증가: 코드의 라인이 줄어들어 핵심 로직에 집중할 수 있습니다.
- 유지 보수성 증가: 프로퍼티 추가, 수정 시 관련 메서드는 자동 생성 됩니다.

MyLittleChicken committed:
refactor: lombok 적용

1. lombok 의존성 추가
2. ShortenUrlEntity 클래스의 getter, constructor 에 적용

Hyune-c commented on src/main/java/community/whatever/onembackendjava/entity/ShortenUrlEntity.java:
```diff
@@ -2,21 +2,27 @@
 import java.util.Optional;
 
+@Getter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
```
NoArgsConstructor 이게 왜 들어간 건가요? 예상되는 것으로는..
- jpa entity 로의 변경을 대비해서? 근데 그때 넣으면 되는거 아닌가요?

그리고
- jpa entity 로 언제 변화하나요? 
- jpa 를 써야되나요? mybatis 를 써야 되면요? 아예 nosql 로 바뀌면요?
- 영속화 책임이 외부 연동으로 바뀌면요?

MyLittleChicken replied:
@Hyune-c 
>NoArgsConstructor 이게 왜 들어간 건가요? 예상되는 것으로는..
>- jpa entity 로의 변경을 대비해서? 근데 그때 넣으면 되는거 아닌가요?
- jpa entity 를 의식해서 적용한 것은 아닙니다. 
- 도메인 로직을 포함하는 객체로써 해당 클래스를 설계했습니다. 영속화의 대상이 아닙니다.

> jpa entity 로 언제 변화하나요?
- jpa entity 로의 변화는 jpa 를 도입할 때 고려하겠습니다.

> jpa 를 써야되나요? mybatis 를 써야 되면요? 아예 nosql 로 바뀌면요?
- UrlShortenRepository 인터페이스의 구현체를 각 스펙에 맞게 구현하면 됩니다.

> 영속화 책임이 외부 연동으로 바뀌면요?
- 상기 내용과 동일합니다.

Hyune-c commented on src/main/java/community/whatever/onembackendjava/entity/ShortenUrlEntity.java:
```diff
@@ -27,41 +33,6 @@ public ShortenUrlEntity(
         this.expiredAt = null;
     }
```
`nit` createEntity 메서드를 위한 생성자를 따로 만드셨네요. 저는 기본 생성자를 재활용할 것 같긴 합니다.
- 코틀린은 named parameter 로 이런게 간결해짐
- 저번에도 비슷한 얘기를 했던거 같은데 저는 record 를 사용할 것 같습니다.

MyLittleChicken replied:
@Hyune-c
>createEntity 메서드를 위한 생성자를 따로 만드셨네요. 저는 기본 생성자를 재활용할 것 같긴 합니다.
- 네 공감합니다. 해당 메서드를 위한 생성자는 제거 하겠습니다.
- 대신 모든 필드를 인자로 받는 생성자를 활용하도록 수정하겠습니다.

>저번에도 비슷한 얘기를 했던거 같은데 저는 record 를 사용할 것 같습니다.
- 네 기억하고 있습니다. 다만 굳이 전환 해야할 이유를 아직 찾지 못한 상태입니다.
- 다양한 케이스에 대한 학습을 좀 더 진행 해보겠습니다.

MyLittleChicken committed:
refactor: 도메인 엔티티와 영속성 엔티티 통합 및 불필요한 기능 제거

1. 현재 단계에서는 굳이 둘을 분리 할 필요가 없어 영속성 엔티티는 제거
2. 만료시간 갱신 기능 제거
3. 삭제 처리 제거
4. 더 이상 mutable 할 필요가 없어, record 로의 전환

MyLittleChicken committed:
refactor: 빈 주입을 위한 생성자를 @RequiredArgsConstructor 로 대체

MyLittleChicken commented on src/main/java/community/whatever/onembackendjava/infrastructure/ShortenUrlRecord.java:
- 엔티티 분리가 현재 단계에서는 의미 없다 판단하여 제거했습니다.

MyLittleChicken commented on src/main/java/community/whatever/onembackendjava/entity/ShortenUrlEntity.java:
- 불필요한 기능을 제거 했습니다.
    - 단축된 url 을 삭제, 만료 기간을 갱신하기 위해 미리 정의한 함수를 제거 했습니다.
    - 그로 인해 엔티티는 더 이상 mutable 할 필요가 없으므로 `record` 로 변경 했습니다.

MyLittleChicken commented on src/main/java/community/whatever/onembackendjava/application/UrlShortenServiceImpl.java:
```diff
@@ -5,41 +5,30 @@
 import java.net.URI;
 
 @Service
+@RequiredArgsConstructor
```
- bean 주입을 위한 생성자는 lombok 에 의해 생성되도록 수정했습니다.
